### ISSUE TITLE
Does the plugin works for higher version of Jekyll?

### ISSUE THREAD

**Issue by garyng** [RobertDeRose/jekyll-livereload#18]
Does the plugin works for higher version of Jekyll?
[github-pages](https://pages.github.com/versions/) now uses jekyll `3.6.2`, but I was unable to make this plugin works:

```
Unable to load the EventMachine C extension; To use the pure-ruby reactor, require 'em/pure_ruby'
C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.5-x64-mingw32/lib/rubyeventmachine.rb:2:in `require': cannot load such file -- 2.4/rubyeventmachine (LoadError)
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.5-x64-mingw32/lib/rubyeventmachine.rb:2:in `<top (required)>'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.5-x64-mingw32/lib/eventmachine.rb:8:in `require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.5-x64-mingw32/lib/eventmachine.rb:8:in `<top (required)>'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/em-websocket-0.5.1/lib/em-websocket.rb:3:in `require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/em-websocket-0.5.1/lib/em-websocket.rb:3:in `<top (required)>'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-livereload-0.2.2/lib/jekyll-livereload/websocket.rb:28:in `require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-livereload-0.2.2/lib/jekyll-livereload/websocket.rb:28:in `<top (required)>'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-livereload-0.2.2/lib/jekyll-livereload.rb:18:in `require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-livereload-0.2.2/lib/jekyll-livereload.rb:18:in `<module:Livereload>'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-livereload-0.2.2/lib/jekyll-livereload.rb:4:in `<module:Jekyll>'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-livereload-0.2.2/lib/jekyll-livereload.rb:3:in `<top (required)>'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:81:in `require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:81:in `block (2 levels) in require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:76:in `each'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:76:in `block in require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:65:in `each'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:65:in `require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler.rb:114:in `require'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-3.6.2/lib/jekyll/plugin_manager.rb:51:in `require_from_bundler'
        from C:/tools/ruby24/lib/ruby/gems/2.4.0/gems/jekyll-3.6.2/exe/jekyll:11:in `<top (required)>'
        from C:/tools/ruby24/bin/jekyll:23:in `load'
        from C:/tools/ruby24/bin/jekyll:23:in `<main>'
```

Any solution?
Thanks!

**Comment by RobertDeRose** [comment]
This is my Gemfile

```ruby
source 'https://rubygems.org'
ruby RUBY_VERSION

gem 'jekyll', '~> 3.6.2'
gem 'bootstrap-sass', '~> 3.3'
gem 'font-awesome-sass', '~> 4.7'
gem 'uglifier', '~> 3.0'
gem 'twitter', '~> 5.11'

group :jekyll_plugins do
  gem 'jekyll-livereload', '~> 0.2'
  gem 'jekyll-paginate', '~> 1.1'
  gem 'jekyll-assets', '~> 2.3'
  gem 'jekyll-sitemap', '~> 0.11'
  gem 'jekyll-admin'
  gem 'jekyll-seo-tag'
  gem 'jekyll-feed', '~> 0.6'
end
```

and 3.6.2 works for me, could I get some more information?

OS, OS version, Ruby version?

I'm on macOS and I have a compiler available so event-machine can install and built, you might need to follow the error's recommendations and use the pure ruby version if you don't have a compiler available

**Comment by garyng** [comment]
Windows 10 
Ruby 2.4.2.2 with msys2

**Comment by RobertDeRose** [comment]
So, you don't have gcc installed? Have you tried using the `em/pure_ruby` like the error reported

**Comment by garyng** [comment]
I guess Windows user have to use `msys2` (instead of `gcc`) for native extensions.. I might be wrong..
For the `em/pure_ruby`, how should I `use`?

I'm lost... Getting Ruby to work on Windows is so tough...

**Comment by RobertDeRose** [comment]
Yeah man, not a Windows user, I'm sorry for your trouble, it's the reason I don't do Windows.  Have you tried the new Linux Subsystem for Windows feature in windows 10, you can then go on the Windows Store and install an [Ubuntu subsystem](https://www.windowscentral.com/how-install-bash-shell-command-line-windows-10).

Anyways you would simply add the following to you Gemfile for the em/pure-ruby

```ruby
source 'https://rubygems.org'
ruby RUBY_VERSION

gem 'jekyll', '~> 3.6.2'

# On Windows, without a compiler, use this
gem 'em/pure_ruby'

gem 'bootstrap-sass', '~> 3.3'
gem 'font-awesome-sass', '~> 4.7'
gem 'uglifier', '~> 3.0'
gem 'twitter', '~> 5.11'

group :jekyll_plugins do
  gem 'jekyll-livereload', '~> 0.2'
  gem 'jekyll-paginate', '~> 1.1'
  gem 'jekyll-assets', '~> 2.3'
  gem 'jekyll-sitemap', '~> 0.11'
  gem 'jekyll-admin'
  gem 'jekyll-seo-tag'
  gem 'jekyll-feed', '~> 0.6'
end
```

**Comment by RobertDeRose** [comment]
Well just tested the above, it doesn't actually seem to work, might need to actually look at this error in more detail.

**Comment by RobertDeRose** [comment]
@garyng Sorry man, so it looks like you have to install `eventmachine` with the C extension for `em-websocket` to work. Which I didn't write, I just depend on it. I would try the Ubuntu Linux Subsystem if you really need to run things like Ruby on Windows. It's gotta be a better experience.

I haven't had to use Windows or wanted to, since 2001, so, my apologies for not knowing too much about that environment. Haven't done any testing of this plugin on Windows to be 100% honest.

**Comment by garyng** [comment]
Daily Windows user struggle indeed... 😢

**Comment by garyng** [comment]
I found this on `eventmachine` original repo: https://github.com/eventmachine/eventmachine/issues/806#issuecomment-332273620

```
...
Note that Windows binaries are not available for Ruby 2.4 yet, in that case you'll need to compile yourself, 
which is easy with: gem install eventmachine --platform ruby
```
Doing 
```
gem uninstall eventmachine
gem install eventmachine --platform ruby
```
makes the plugin works again on Windows
(from https://github.com/eventmachine/eventmachine/issues/806#issuecomment-338863970)

Cheers! 🍺 

Thanks again! 😄
